### PR TITLE
Handle QueryBuilder requests

### PR DESCRIPTION
**Pull request by travisstebbins**
Add LSP functionality to handle the basic requests needed by QueryBuilder, including
- `getClassifierPathMapping`
- `getSubtypeInfo`
- `transformLambdasToCode`
- `transformLambdaToCode`
- `transformCodeToLambda`
- `getLambdaReturnType`
- `runQuery`
- `exportData`
- `generateExecutionPlan`
- `debugExecutionPlanGeneration`
- `analyzeMappingModelCoverage`

### CODE REVIEW

File: legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/mapping/MappingLSPGrammarExtension.java [hunk 1]
```diff
@@ -264,6 +250,74 @@ private List<? extends LegendExecutionResult> runLegacyMappingTests(SectionState
...
+        CompileResult compileResult = this.getCompileResult(section);
+
+        if (compileResult.hasException())
+        {
+            return Collections.singletonList(errorResult(compileResult.getCompileErrorResult(), entityPath));
+        }
+
+        try
+        {
+            if (this.isEngineServerConfigured()
```
Comment (rafaelbey) [inline legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/mapping/MappingLSPGrammarExtension.java:265]: I will avoid this if, as this is mostly static code, and does not required authentication to run, hence running always locally would make sense.

Function execution is one of the few things we dispatch to the server as connecting to some Databases is not possible yet in LSP.

File: legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java [hunk 2]
```diff
@@ -192,6 +309,181 @@ static void executePlan(GlobalState globalState, FunctionExecutionSupport execut
...
+            );
+        }
+        catch (Exception e)
+        {
+            results.add(extension.errorResult(e, entityPath));
+        }
+        return results;
+    }
+
+    static Iterable<? extends LegendExecutionResult> convertJSONToGrammar_lambda_batch(FunctionExecutionSupport executionSupport, SectionState section, String entityPath, Map<String, String> executableArgs, Map<String, Object> inputParameters)
```
Comment (rafaelbey) [inline legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java:418]: Lets follow Java naming conventions:

`convertLambdasJsonToGrammar`

File: legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java [hunk 3]
```diff
@@ -192,6 +309,181 @@ static void executePlan(GlobalState globalState, FunctionExecutionSupport execut
...
+            );
+        }
+        catch (Exception e)
+        {
+            results.add(extension.errorResult(e, entityPath));
+        }
+        return results;
+    }
+
+    static Iterable<? extends LegendExecutionResult> convertJSONToGrammar_lambda(FunctionExecutionSupport executionSupport, SectionState section, String entityPath, Map<String, String> executableArgs, Map<String, Object> inputParameters)
```
Comment (rafaelbey) [inline legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java:389]: Do we need this method and the one below?  Can we just use the bulk even for single calls?

If we end needing this method, can we delegate to the bulk one (or vice-versa) and avoid some duplication?  Also, lets name it following Java convention: `convertLambdaJsonToGrammar`

File: legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java [hunk 4]
```diff
@@ -192,6 +309,181 @@ static void executePlan(GlobalState globalState, FunctionExecutionSupport execut
...
+            );
+        }
+        catch (Exception e)
+        {
+            results.add(extension.errorResult(e, entityPath));
+        }
+        return results;
+    }
+
+    static Iterable<? extends LegendExecutionResult> convertGrammarToJSON_lambda(FunctionExecutionSupport executionSupport, SectionState section, String entityPath, Map<String, String> executableArgs, Map<String, Object> inputParameters)
```
Comment (rafaelbey) [inline legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java:358]: Java naming convention: `convertLambdaGrammarToJson`

File: legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/mapping/MappingModelCoverageAnalysisRequest.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.finos.legend.engine.ide.lsp.extension.mapping;
+
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContext;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContextData;
+
+public class MappingModelCoverageAnalysisRequest
```
Comment (rafaelbey) [inline legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/mapping/MappingModelCoverageAnalysisRequest.java:22]: Think if we remove the remote call, we can also remove this class?

**Review by rafaelbey — APPROVED** [review]

File: legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/mapping/MappingLSPGrammarExtension.java (hunk 1 above)
Comment (travisstebbins) [reply legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/mapping/MappingLSPGrammarExtension.java]: Makes sense. I wasn't sure if we wanted all functionality to be able to be done remotely, so I added this. But if we don't need to support everything remotely, then we don't need this, so I removed it.

File: legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java (hunk 2 above)
Comment (travisstebbins) [reply legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java]: Done

File: legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java (hunk 3 above)
Comment (travisstebbins) [reply legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java]: Good point. I removed the single version and kept only the batch version. The client will just call the batch version with a map with one element if it only needs to convert 1 JSON.

File: legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java (hunk 4 above)
Comment (travisstebbins) [reply legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java]: Done

File: legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/mapping/MappingModelCoverageAnalysisRequest.java (hunk 5 above)
Comment (travisstebbins) [reply legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/mapping/MappingModelCoverageAnalysisRequest.java]: Yep, this isn't needed anymore so I removed it.

**Review by rafaelbey — APPROVED** [review]

**Pull request merged**